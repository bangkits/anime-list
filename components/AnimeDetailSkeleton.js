/** @jsxImportSource @emotion/react */
import { Skeleton, Grid } from "@mui/material"
import AniToolBar from "./AniToolBar";

const AnimeDetailSkeleton = () => {
  return (
    <div css={{ marginTop: 90 }}>
      <AniToolBar hasBackButton title="Anime Detail" />
      <Grid
        container
        css={{
          marginTop: 40,
          padding: 30,
          '@media(min-width: 768px)': {
            paddingLeft: 100
          }
        }}
      >
        <Grid item xs={12} md={6} lg={4}>
          <Skeleton
            variant="rectangular"
            height="60vh"
            animation="wave"
            css={{
              '@media(min-width: 768px)': {
                width: 375
              }
            }}
          />
        </Grid>
        <Grid item xs={12} md={6} lg={6}>
          <Skeleton animation="wave" height={75} />
          <div css={{ display: 'flex', gap: 5 }}>
            {
              [...Array(5).keys()].map((v, i) => <Skeleton key={i} animation="wave" variant="circular" width={18} />)
            }
          </div>
          <Skeleton width="30%" />
          <Skeleton width="30%" />
          <Skeleton width="30%" height="10vh" />
          <Skeleton height="30vh" />
        </Grid>
      </Grid>
    </div>
  )
};

export default AnimeDetailSkeleton;