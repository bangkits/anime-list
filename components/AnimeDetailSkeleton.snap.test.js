import renderer from 'react-test-renderer';
import AnimeDetailSkeleton from './AnimeDetailSkeleton';

jest.mock('./AniToolBar', () => 'AniToolBar');

describe('AnimeDetailSkeleton Component Snapshot', () => {
  it('Renders AnimeDetailSkeleton component input', () => {
    const tree = renderer
      .create(<AnimeDetailSkeleton />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
