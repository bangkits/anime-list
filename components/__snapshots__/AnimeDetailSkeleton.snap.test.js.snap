// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeDetailSkeleton Component Snapshot Renders AnimeDetailSkeleton component input 1`] = `
<div
  className="css-16ca04s-AnimeDetailSkeleton"
>
  <AniToolBar
    hasBackButton={true}
    title="Anime Detail"
  />
  <div
    className="MuiGrid-root MuiGrid-container css-3fzui7-MuiGrid-root-AnimeDetailSkeleton"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-1e6jbaj-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-i7obss-MuiSkeleton-root-AnimeDetailSkeleton"
        style={
          Object {
            "height": "60vh",
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-1fyyp8j-MuiGrid-root"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
        style={
          Object {
            "height": 75,
            "width": undefined,
          }
        }
      />
      <div
        className="css-st3jg2-AnimeDetailSkeleton"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-twt492-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": 18,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-twt492-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": 18,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-twt492-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": 18,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-twt492-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": 18,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-twt492-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": 18,
            }
          }
        />
      </div>
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": undefined,
            "width": "30%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": undefined,
            "width": "30%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": "10vh",
            "width": "30%",
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
        style={
          Object {
            "height": "30vh",
            "width": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;
